src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/summary_tab.cljs

```diff
@@ -107,6 +115,18 @@
                    (if success?
                      ((:on-delete props))
                      (js/alert (str "Error on delete: " status-text))))}))
+   :lock-or-unlock
+   (fn [{:keys [props state]} locked-now?]
+     (swap! state assoc :locking? locked-now?)
+     (endpoints/call-ajax-orch
+       {:endpoint (endpoints/lock-or-unlock-workspace (:workspace-id props) locked-now?)
+        :on-done (fn [{:keys [success? status-text status-code]}]
+                   (if success?
+                     (swap! state dissoc :server-response)
+                     (if (and (= status-code 409) (not locked-now?))
+                       (js/alert "Could not lock workspace, one or more analyses are currently running")
+                       (js/alert (str "Error: " status-text))))
+                   (swap! state dissoc :locking?))}))
```

Comment:
Each one of these swaps causes a component-did-update, which reloads the workspace. Try:
```Clojure
(swap! state #(if success? (dissoc % :locking? :server-response) (dissoc % :locking?)))
```